xds/internal/xdsclient/xdsresource/unmarshal_cds_test.go

```diff
@@ -1620,3 +1622,181 @@ func (s) TestUnmarshalCluster(t *testing.T) {
+			OutlierDetection: od,
+		}
+	}
+	odUpdate := func(od *OutlierDetection) ClusterUpdate {
+		return ClusterUpdate{
+			ClusterName:      clusterName,
+			LRSServerConfig:  ClusterLRSOff,
+			OutlierDetection: od,
+		}
+	}
+
+	odDefaultUpdate := &OutlierDetection{
+		Interval:                       10 * time.Second,
+		BaseEjectionTime:               30 * time.Second,
+		MaxEjectionTime:                300 * time.Second,
+		MaxEjectionPercent:             10,
+		SuccessRateStdevFactor:         1900,
+		EnforcingSuccessRate:           100,
+		SuccessRateMinimumHosts:        5,
+		SuccessRateRequestVolume:       100,
+		FailurePercentageThreshold:     85,
+		EnforcingFailurePercentage:     0,
+		FailurePercentageMinimumHosts:  5,
+		FailurePercentageRequestVolume: 50,
+	}
+
+	odProtoWithValidFields := &v3clusterpb.OutlierDetection{
+		Interval:                       &durationpb.Duration{Seconds: 1},
+		BaseEjectionTime:               &durationpb.Duration{Seconds: 2},
+		MaxEjectionTime:                &durationpb.Duration{Seconds: 3},
+		MaxEjectionPercent:             &wrapperspb.UInt32Value{Value: 1},
+		SuccessRateStdevFactor:         &wrapperspb.UInt32Value{Value: 2},
+		EnforcingSuccessRate:           &wrapperspb.UInt32Value{Value: 3},
+		SuccessRateMinimumHosts:        &wrapperspb.UInt32Value{Value: 4},
+		SuccessRateRequestVolume:       &wrapperspb.UInt32Value{Value: 5},
+		FailurePercentageThreshold:     &wrapperspb.UInt32Value{Value: 6},
+		EnforcingFailurePercentage:     &wrapperspb.UInt32Value{Value: 7},
+		FailurePercentageMinimumHosts:  &wrapperspb.UInt32Value{Value: 8},
+		FailurePercentageRequestVolume: &wrapperspb.UInt32Value{Value: 9},
+	}
+	odUpdateWithValidFields := &OutlierDetection{
+		Interval:                       time.Second,
+		BaseEjectionTime:               time.Second * 2,
+		MaxEjectionTime:                time.Second * 3,
+		MaxEjectionPercent:             1,
+		SuccessRateStdevFactor:         2,
+		EnforcingSuccessRate:           3,
+		SuccessRateMinimumHosts:        4,
+		SuccessRateRequestVolume:       5,
+		FailurePercentageThreshold:     6,
+		EnforcingFailurePercentage:     7,
+		FailurePercentageMinimumHosts:  8,
+		FailurePercentageRequestVolume: 9,
+	}
+
+	tests := []struct {
+		name       string
+		cluster    *v3clusterpb.Cluster
+		wantUpdate ClusterUpdate
+		wantErr    bool
+	}{
+		{
+			name: "successful-case-all-defaults",
+			// Outlier detection proto is present without any fields specified,
+			// so should trigger all default values in the update.
+			cluster:    odProto(&v3clusterpb.OutlierDetection{}),
+			wantUpdate: odUpdate(odDefaultUpdate),
+		},
+		{
+			name:       "successful-case-all-fields-configured-and-valid",
+			cluster:    odProto(odProtoWithValidFields),
+			wantUpdate: odUpdate(odUpdateWithValidFields),
+		},
+		{
+			name: "interval-is-negative",
+			cluster: odProto(&v3clusterpb.OutlierDetection{
+				Interval: &durationpb.Duration{Seconds: -10},
```

Comment:
Totally optional: If you want to save a line (and sometimes I definitely find that useful since you scroll less when you are reading the code), you can specify the `cluster` field in a single line since the `OutlierDetection` proto has a single field.

Reply:
I preferred my way for this, but I switched anyway :D.